README.md
# Tarea 1: Fundamentos Devops 2023 Otoño

Ejecute los siguientes pasos:

1. Haga un fork de este repo
2. Agregue un archivo, póngale su nombre o use su identificador de usuario en Github para nombrarlo
3. En el interior de este archivo escriba su nombre completo
4. Haga commit con un mensaje al estilo "commit convencional" de tipo "feat" (ejemplo: `git commit -am "feat: agrega archivo solicitado"`)
5. Ejecute el comando `git log` y agregue al archivo anterior el contenido asociado al commit ejecutado en el paso 4 (tip: mire el archivo con el nombre del profesor en este repo para ver lo que se espera que contenga)
6. Haga un nuevo commit (`git commit -am "feat: agrega contenido del git log"`)
7. Modifique el archivo `README.md` agregando su nombre y la fecha en la tabla que está al final de este archivo.
8. Haga un nuevo commit convencional de tipo fix (ejemplo: `git commit -am "fix: agregué mi nombre: Eduardo Diaz Cortes a README.md")
9. Ejecute el commando `git blame README.md` y busque la primera linea que contenga su nombre, agregue el código que sale a la izquierda a la tabla que está al final de este archivo (al lado de su nombre).
10. Haga un nuevo commit
11. ¿Qué hace el comando `git blame README.md`? Agregue su respuesta en el archivo que usted creó en el paso 2.
12. Haga un nuevo commit
13. En el archivo CHANGELOG.md agregue la fecha y el detalle de los cambios que ha realizado
14. Haga un nuevo commit
15. Si estuvo trabajando en su PC asegúrese de hacer un push a su fork (`git push origin main`)
16. Realize un Pull Request hacia este repo, esa será la forma en que entregará su tarea
17. Fíjese que en este repo hay un ejemplo ejecutado por el profesor, por si tiene dudas de cómo realizar la tarea
18. Si las dudas persisten use el foro de u-cursos para realizar sus consultas.



# Tabla de Respuestas

Deje acá lo solicitado en los pasos 7 y 9

| Nombre        | Fecha               | Blame Hash |
| ------------- | ------------------- | ---------- |
| Eduardo Diaz  | 14 de julio de 2023 | 73f38606   |
|Carlos Torres|14 de julio de 2023|9117d62|
|Paula Riffo|15 de julio de 2023|b8dee6d0|
| Diego Jiménez | 15 de julio de 2023 | 05c1eea5   |
|Ignacio Tomasov Silva|16 de julio de 2023|3a33fa56|
|Seba Gonzalez|17 de julio de 2023|b03017c9|
|Marbely Graterol|17 de julio de 2023|03e84b9a|
|Diego Sepúlveda Soto|19 de julio de 2023|200351b7|
|Rodrigo Tobar|20 de julio de 2023|e4a21cb2|
|Johann Llanos|20 de julio de 2023|f02af4ac|
|Luis Díaz|20 de Julio de 2023|0cca998f|
|Eduardo Galaz|20 de julio de 2023|e0f8cc2f|
|Patricia Campos|21 de julio de 2023|4dde1e06|
|Melquisedec Sierra|21 de julio de 2023|6d2b3648|
|Victor Vega |07 de julio de 2023|2d9117c8|
|Fabian Castillo|23 de julio de 2023|1b011878|
|Esteban Gonzalez|23 julio de 2023|400e2124|
|Cesar Soto|24 de julio de 2023|b9931c76|
|David Carvallo|24 de julio de 2023|67ef9cd0|
|Rodrigo Reyes|24 de julio de 2023|a7809cb7|
|Gabriel Aravena|24 de julio de 2023|11243112|


CHANGELOG.md
# Historial de cambios

2023/07/14: Agrega datos del profe

2023/07/24: Se agrega el detalle de los cambios realizados

commit 2926b3f44c0bd6a37060628615065c424f7915c2 (HEAD -> main)
Author: GabrielAravenaRivero <gabriel.aravenar@apoyate.cl>
Date:   Mon Jul 24 23:34:38 2023 -0400

    fix: agregué la respuesta a la pregunta del punto 11

commit 7ce8059f034af425ad6147f55ce9b363ed071a1b
Author: GabrielAravenaRivero <gabriel.aravenar@apoyate.cl>
Date:   Mon Jul 24 23:29:53 2023 -0400

    fix: agregué el hash 11243112 a README.md

commit 1124311260ffde95040a99d1a95503a398c6b009
Author: GabrielAravenaRivero <gabriel.aravenar@apoyate.cl>
Date:   Mon Jul 24 23:27:57 2023 -0400

    fix: agregué mi nombre: Gabriel Aravena Rivero a README.md

commit 63b2dfd7c303e727db6cbe921f87299fd7fcfaa1
Author: GabrielAravenaRivero <gabriel.aravenar@apoyate.cl>
Date:   Mon Jul 24 23:25:42 2023 -0400

    feat: agrega contenido del git log

commit c86968b76ced46932523984c1803de0c5f313ed1 (origin/main, origin/HEAD)
Author: Gabriel Aravena <64670545+GabrielAravena@users.noreply.github.com>
Date:   Mon Jul 24 23:19:15 2023 -0400

    feat: agrega archivo solicitado
- 2023/07/24: Agrego cambios Tarea 1

- 2023/07/21: Se agregaron datos y respuesta a preguntas de tarea por parte de David Carvallo.

- 2023/07/24: se crea archivo cesarsotoba

- 2023/07/24: se agrega resultado de git log a archivo cesarsotoba

- 2023/07/24: se agrega fila con fecha y nombre a grilla en archivo README.md

- 2023/07/24: se obtiene código del cambio en línea del archivo README.md con el comando git blame

- 2023/07/24: se actualiza grilla en archivo README.md con el código de cambio de línea. 

- 2023/07/24: se agrega definición de git blame en archivo cesarsotoba

- 2023/07/24: se actualiza y cambia extension del archivo cesarsotoba a cesarsotoba.md

- 2023/07/23 : hice la tarea del profe, agregue un archivo egon27.md y realice modificaciones al archivo README.md

- 2023/07/23: Agrega datos de Fabian Castillo
- 2023/07/23: Victor Vega <Victor.Vega.Barrientos@gmail.com> 

### Added

- Nuevo archivo "victorVega.java" que contiene lo solicitado en la tarea

### Changed

- Se modifica archivo "README.md" con la fecha, mi nombre y el hash obtenido con git blame.
- Se modifica el archivo "victorVega.java" con la respuesta solicitada en la tarea.

## [1.0.0] - 2023-07-21

### Added

- Se agregan datos de Eduardo Galaz en eduardo.galaz.md
- Se agrega información de git log en eduardo.galaz.md
- Se agrega el nombre de Eduardo Galaz en README.md
- Se agrega el identificador de commit de Eduardo Galaz en README.md
- Se agrega información de git blame en eduardo.galaz.md

- 2023/07/21: Agrega datos del estudiante Melquisedec Sierra
-	Adicion de datos en tabla correspondientes a nombre, fecha, commit[:8]
- Respuesta pregunta 11 sobre blame
- 2023/07/21: Se agrega Archivo PatriciaCampos.md, se cambia README.md y se responden preguntas de acuerdo a instrucciones de la tarea.
- 2023/07/20: Se agregan datos de Luis Diaz, se modifica contenido del CHANGELOG y se añade entrada a tabla en README.
- 2023/07/20: Agrega archivo jllanosf.md y datos tabla en README.md
- 2023/07/20: Ingresa archivos y modificaciones para la tarea de Rodrigo Tobar.
- 2023/07/19: Se agregan detalles del almuno Diego Sepúlveda, documentando lo solicitado
- 2023/07/17: Agrega datos del alumno, y agrega elementos solicitados.
- 2023/07/17: Agregar archivo solicitado (marbely.graterol.md)
- 2023/07/17: Agregar contenido del git log
- 2023/07/17: Agregar nombre: Marbely Graterol a README.md
- 2023/07/17: Agregar agregar código blame a README.md
- 2023/07/17: Agregar documentación de cómo funciona el comando blame
- 2023/07/16: Ignacio Tomasov agrega su tarea
- 2023/07/15: Agrega datos Paula Riffo
- 2023/07/15: Agrega resultado `git log`
- 2023/07/15: Agrega respuesta a pregunta `¿Qué hace el comando git blame README.md?`
- 2023/07/14: Se agrega un nuevo archivo "CharlyTango21.md" y se modifica el archivo "README.md"
- 2023/07/15: Agrega los cambios solicitados para el cumplimiento de la tarea
- 2023/07/15: Agregar archivo djimenezpalacios12.md
- 2023/07/15: Agregar contenido solicitado al archivo djimenezpalacios12.md
- 2023/07/15: Modificación del archivo README.md
- 2023/07/15: Agregar contenido para responder pregunta sobre el commando git blame>>>>>>> main
- 2023/07/14: Agrega datos del profe


victorVega.java
//Nombre: Victor Sebastián Vega Barrientos

/*
--------------------GIT LOG---------------------: 


commit 8bf7d7381eaba2fba2a783972668f15667b8e5c6 (HEAD -> main)
Author: Replit user <>
Date:   Sun Jul 23 05:38:49 2023 +0000

    feat: se incluye nuevo archivo solicitado

commit 8964cb354e411bed07a2f1b2dffe6aa16830dae2 (origin/main)
Author: Eduardo Díaz <ediaz@dcc8090.cl>
Date:   Fri Jul 14 13:32:08 2023 -0400

    Update CHANGELOG.md

commit da7cd096b5c2e4d1ec5ddcc08c6c8f73cfa896fc
Author: Eduardo Diaz <ediaz@dcc8090.cl>
Date:   Fri Jul 14 13:30:33 2023 -0400

    feat: actualiza el blame

commit 10ae9d3549f9f343e736d36f34ef781d65ba6e9e
Author: Eduardo Diaz <ediaz@dcc8090.cl>
Date:   Fri Jul 14 13:26:13 2023 -0400

    feat: agrega archivo solicitado

commit 4b30cce691eff19e06abd54c8bee513e36b24bea
Author: Eduardo Díaz <ediaz@dcc8090.cl>
Date:   Fri Jul 14 13:28:54 2023 -0400

    Update README.md

commit e09dbd3542dab28621b6693cd154a018fb3130eb
Author: Eduardo Díaz <ediaz@dcc8090.cl>
Date:   Fri Jul 14 13:25:13 2023 -0400

    Create CHANGELOG.md

commit 73f3860691eca49bc0ce2bba1f16f164f13650d6
Author: Eduardo Díaz <ediaz@dcc8090.cl>
Date:   Fri Jul 14 13:20:01 2023 -0400

    Update README.md

commit 9b6ce51233265fe7b50d8e3c09ed89c9a8c8bb56
Author: Eduardo Díaz <ediaz@dcc8090.cl>
Date:   Fri Jul 14 13:15:23 2023 -0400


-------------------FIN GIT LOG------------------------

Pregunta y respuesta punto 11 : ¿Qué hace el comando git blame README.md?

R: Este comando sirve para revisar lo que contiene un archivo, con el detalle de cada modificación y la información del autor de cada linea.

*/